scripts.js

```diff
@@ -0,0 +1,226 @@
+    this.previous.addEventListener("click", this.switchPreviousQuestion.bind(this));
+    this.buttonName.addEventListener("click", this.eventButtonName.bind(this));
+    this.restart.addEventListener("click", this.eventRestart.bind(this));
+  }
+
+  checkResult(event) {
+    let target = event.target;
+    if (target.tagName != 'LABEL') return;
+
+    this.disableButton();
+
+    let answer = target.previousElementSibling.value;
+    let isCorrect = this.correctDefaultAnswers.includes(answer) || this.correctRequestAnswers.includes(answer);
+    if (isCorrect) {
+      target.style.background = "green";
+      this.countAnswers++;
+    } else {
+      target.style.background = "red";
+    }
+
+    clearTimeout(this.timerShowQuestion);
+
+    this.timerCheckQuestion = setTimeout(this.switchNextQuestion.bind(this), 1500);
+  }
+
+  disableButton() {
+    const disableBtn = this.slides[this.countSlide].querySelectorAll('.radio');
+    Array.from(disableBtn).forEach(item => item.setAttribute("disabled", "disabled"));
+
+    const disableAnswers = this.slides[this.countSlide].querySelector('.question__answers');
+    disableAnswers.removeEventListener("click", this.checkResultBind);
+  }
+
+  startGame() {
+    this.start.classList.add('none');
+    this.next.classList.remove('none');
+    this.previous.classList.remove('none');
+    this.switchQuestion(this.countSlide);
+    console.log(`start ${this.countSlide + 1}`);
+  }
+
+  switchQuestion(switchedSlide) {
+    this.slides[this.countSlide].classList.remove('active');
+    this.slides[switchedSlide].classList.add('active');
+
+    if (this.timerShowQuestion !== null) clearTimeout(this.timerShowQuestion);
+    if (this.countSlide !== 3) {
+      this.timerShowQuestion = setTimeout(() => {
+        this.disableButton();
+        this.switchNextQuestion();
+      }, 10000);
+    }
+  }
+
+  switchNextQuestion() {
+    if (this.countSlide === 2) {
+      this.next.textContent = 'check';
+    }
+
+    if (this.countSlide === 3) {
+      this.next.classList.add('none');
+      this.previous.classList.add('none');
+    }
+
+    if (this.countSlide === 4) return;
+
+
+    this.switchQuestion(this.countSlide + 1);
+    this.countSlide++;
+    console.log(`next ${this.countSlide + 1}`);
+    clearTimeout(this.timerCheckQuestion);
+  }
+
+  switchPreviousQuestion() {
+    if (this.countSlide <= 0) {
+      return;
+    }
+
+    if (this.countSlide != 2) {
+      this.next.textContent = 'next';
+    }
+
+    this.switchQuestion(this.countSlide - 1);
+    this.countSlide--;
+    console.log(this.countSlide + 1);
+  }
+
+  eventButtonName() {
+    let name = this.formName.value;
+    let err = document.createElement('p');
+    if (!this.regex.test(name)) {
+      err.classList.add('error');
+      err.textContent = 'Введите корректное имя(содержит 2-10 символов и начинается с заглавной буквы)';
+      this.form.append(err);
+    } else {
+      this.slides[4].classList.remove('active');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/voronkova-ab/game-js/pullRequest?prid=7705696)
```suggestion
      this.slides[4].classList.remove("active");
```